klippy/extras/resonance_tester.py

```diff
@@ -32,49 +32,54 @@ def get_supported_axes(self):
         self.hz_per_sec = gcmd.get_float("HZ_PER_SEC", self.hz_per_sec,
                                          above=0., maxval=2.)
-        # Attempt to adjust maximum acceleration and acceleration to
-        # deceleration based on the maximum test frequency.
-        max_accel = self.freq_end * self.accel_per_hz
-        toolhead.cmd_SET_VELOCITY_LIMIT(self.gcode.create_gcode_command(
-            "SET_VELOCITY_LIMIT", "SET_VELOCITY_LIMIT",
-            {"ACCEL": max_accel, "ACCEL_TO_DECEL": max_accel}))
-    def run_test(self, toolhead, axis, gcmd):
+    def run_test(self, axis, gcmd):
+        toolhead = self.printer.lookup_object('toolhead')
         X, Y, Z, E = toolhead.get_position()
         if axis not in self.get_supported_axes():
             raise gcmd.error("Test axis '%s' is not supported", axis)
         vib_dir = (1, 0) if axis == 'x' else (0., 1.)
         sign = 1.
         freq = self.freq_start
+        # Override maximum acceleration and acceleration to
+        # deceleration based on the maximum test frequency
+        _, old_max_accel = toolhead.get_max_velocity()
+        max_accel = self.freq_end * self.accel_per_hz
+        toolhead.cmd_SET_VELOCITY_LIMIT(self.gcode.create_gcode_command(
+            "SET_VELOCITY_LIMIT", "SET_VELOCITY_LIMIT",
+            {"ACCEL": max_accel, "ACCEL_TO_DECEL": max_accel}))
         input_shaper = self.printer.lookup_object('input_shaper', None)
         if input_shaper is not None and not gcmd.get_int('INPUT_SHAPING', 0):
             input_shaper.disable_shaping()
             gcmd.respond_info("Disabled [input_shaper] for resonance testing")
         else:
             input_shaper = None
         gcmd.respond_info("Testing frequency %.0f Hz" % (freq,))
-        _, max_accel = toolhead.get_max_velocity()
         while freq <= self.freq_end + 0.000001:
             t_seg = .25 / freq
-            accel = min(self.accel_per_hz * freq, max_accel)
-            V = accel * t_seg
+            accel = self.accel_per_hz * freq
+            max_v = accel * t_seg
             toolhead.cmd_M204(self.gcode.create_gcode_command(
                 "M204", "M204", {"S": accel}))
             L = .5 * accel * t_seg**2
             nX = X + sign * vib_dir[0] * L
             nY = Y + sign * vib_dir[1] * L
-            toolhead.move([nX, nY, Z, E], V)
-            toolhead.move([X, Y, Z, E], V)
+            toolhead.move([nX, nY, Z, E], max_v)
+            toolhead.move([X, Y, Z, E], max_v)
             sign = -sign
             old_freq = freq
             freq += 2. * t_seg * self.hz_per_sec
             if math.floor(freq) > math.floor(old_freq):
                 gcmd.respond_info("Testing frequency %.0f Hz" % (freq,))
+        # Restore the original max acceleration value
+        toolhead.cmd_SET_VELOCITY_LIMIT(self.gcode.create_gcode_command(
+            "SET_VELOCITY_LIMIT", "SET_VELOCITY_LIMIT",
+            {"ACCEL": old_max_accel, "ACCEL_TO_DECEL": old_max_accel * .5}))
```

Comment:
As mentioned above, I think it would be preferable to restore max_accel_to_decel.

Reply:
Done.